VocaDb.Migrations/Migrations.cs

```diff
@@ -1,10 +1,157 @@
 using System.Data;
+using System.Linq;
 using FluentMigrator;
 
 namespace VocaDb.Migrations
 {
 	// Migration version format: YYYY_MM_DD_HHmm
 
+	[Migration(2020_11_20_2000)]
+	public class Comments : Migration
+	{
+		private sealed class CommentTable
+		{
+			public CommentTable(string name, string schema = null, string entryTypeName = null)
+			{
+				Name = name;
+				Schema = schema;
+				EntryTypeName = entryTypeName;
+			}
+
+			/// <summary>
+			/// Table name without schema. (e.g. `AlbumComments`, `ArtistComments`, `DiscussionComments`, ...)
+			/// </summary>
+			public string Name { get; }
+
+			/// <summary>
+			/// Table schema. (either null or `discussions`)
+			/// </summary>
+			public string Schema { get; }
+
+			/// <summary>
+			/// Entry type name. (e.g. `Album`, `Artist`, `Topic`, ...)
+			/// </summary>
+			public string EntryTypeName { get; }
+
+			/// <summary>
+			/// Table name with schema. (e.g. `AlbumComments`, `ArtistComments`, `discussions.DiscussionComments`, ...)
+			/// </summary>
+			public string NameWithSchema => Schema != null ? $"{Schema}.{Name}" : Name;
+		}

```

Comment:
I might be too late now, but wouldn't it make more sense to have the above strings be some sort of enumerations (if available), or is type safety ensured elsewhere? There may also be a performance benefit, but then again I suppose indexing takes care of that.

It also looks odd to me to have information duplication in `NameWithSchema`. Shouldn't this filtering be encapsulated in logic, or is this for performance reasons?

(I have no clue about these frameworks, or many best practices, so apologies if my comments seem more like incoherent rambling)

Reply:
The `CommentTable` is for one-time use only. We have 10 different comment tables: `AlbumComments`, `ArtistComments`, 
`discussions.DiscussionComments`, `ReleaseEventComments`, `SongComments`, `SongListComments`, `TagComments`, `UserComments`, `discussions.DiscussionTopics` and `AlbumReviews`. As you can see, `discussions.DiscussionComments` and `discussions.DiscussionTopics` reside in a different schema. So I think this is the easiest way.